### PR TITLE
feat!: Add dates as a top level field to credit batches (#393)

### PR DESCRIPTION
**Pull request by ruhatch**
## Description

Add `start_date` and `end_date` fields to `BatchInfo` and associated `MsgCreateBatchRequest`. These dates track the period during which the credit batch was quantified and verified.

Closes: #393

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/msgs.go [hunk 1]
```diff
@@ -30,6 +30,10 @@ func (m *MsgCreateClassRequest) GetSigners() []sdk.AccAddress {
 }
 
 func (m *MsgCreateBatchRequest) ValidateBasic() error {
+	if m.EndDate.Before(*m.StartDate) {
```
Comment (aaronc) [inline x/ecocredit/msgs.go:33]: We need to also check that the dates are actually provided.

File: x/ecocredit/server/testsuite/suite.go [hunk 2]
```diff
@@ -103,12 +104,29 @@ func (s *IntegrationTestSuite) TestScenario() {
...
 	t0, t1, t2 := "10.37", "1007.3869", "0"
 	tSupply0 := "1017.7569"
-	r0, r1, r2:= "4.286", "10000.4589902", "0"
+	r0, r1, r2 := "4.286", "10000.4589902", "0"
 	rSupply0 := "10004.7449902"
 
+	time1 := time.Now()
+	time2 := time.Now()
+
+	// Batch creation should fail if the EndDate is before the StartDate
```
Comment (aaronc) [inline x/ecocredit/server/testsuite/suite.go:113]: Let's also add some tests cases for where one or both dates are omitted and verify that it fails. We could even just do these against Msg.ValidateBasic

**Review by aaronc — APPROVED** [review]
LGTM

**Review by robert-zaremba — APPROVED** [review]
approving, but left 2 suggestions which will improve the code maintenance.

**Review by likhita-809 — APPROVED** [review]
lgtm

**Pull request merged**